specification/containerinstance/resource-manager/Microsoft.ContainerInstance/preview/2024-10-01-preview/containerInstance.json

```diff
@@ -0,0 +1,2605 @@
+          "type": "number",
+          "format": "double"
+        },
+        "gpu": {
+          "description": "The GPU request of this container instance.",
+          "$ref": "#/definitions/GpuResource"
+        }
+      }
+    },
+    "ResourceLimits": {
+      "description": "The resource limits.",
+      "type": "object",
+      "properties": {
+        "memoryInGB": {
+          "description": "The memory limit in GB of this container instance.",
+          "type": "number",
+          "format": "double"
+        },
+        "cpu": {
+          "description": "The CPU limit of this container instance.",
+          "type": "number",
+          "format": "double"
+        },
+        "gpu": {
+          "description": "The GPU limit of this container instance.",
+          "$ref": "#/definitions/GpuResource"
+        }
+      }
+    },
+    "GpuResource": {
+      "description": "The GPU resource.",
+      "type": "object",
+      "required": [
+        "count",
+        "sku"
+      ],
+      "properties": {
+        "count": {
+          "description": "The count of the GPU resource.",
+          "type": "integer",
+          "format": "int32"
+        },
+        "sku": {
+          "type": "string",
+          "description": "The SKU of the GPU resource.",
+          "enum": [
+            "K80",
+            "P100",
+            "V100"
+          ],
+          "x-ms-enum": {
+            "name": "GpuSku",
+            "modelAsString": true
+          }
+        }
+      }
+    },
+    "AzureFileVolume": {
+      "description": "The properties of the Azure File volume. Azure File shares are mounted as volumes.",
+      "type": "object",
+      "required": [
+        "shareName",
+        "storageAccountName"
+      ],
+      "properties": {
+        "shareName": {
+          "description": "The name of the Azure File share to be mounted as a volume.",
+          "type": "string"
+        },
+        "readOnly": {
+          "description": "The flag indicating whether the Azure File shared mounted as a volume is read-only.",
+          "type": "boolean"
+        },
+        "storageAccountName": {
+          "description": "The name of the storage account that contains the Azure File share.",
+          "type": "string"
+        },
+        "storageAccountKey": {
+          "description": "The storage account access key used to access the Azure File share.",
+          "type": "string"
+        },
+        "storageAccountKeyReference": {
+          "description": "The reference to the storage account access key used to access the Azure File share.",
+          "type": "string"
+        }
+      }
+    },
+    "EmptyDirVolume": {
+      "description": "The empty directory volume.",
+      "type": "object",
+      "properties": {}
+    },
+    "SecretVolume": {
+      "description": "The secret volume.",
+      "type": "object",
+      "additionalProperties": {
+        "type": "string"
+      }
+    },
+    "SecretReferenceVolume": {
```

Comment:
>SecretReferenceVolume [](http://example.com/codeflow?start=5&length=21)

how is this property used?

Reply:
This is because ARM is trying to reduce instances of unschematized APIs to make it easier for consumers of those APIs to understand them and use them reliably

It is recommended to use arrays with pairs in place of dictionary if defining a schema is not possible. 

If it is not possible provide an explanation for why you need this and obtain a suppresion for the Swagger LintDiff validation.